### DISCUSSION TITLE
nixos shared directories

### DISCUSSION THREAD

**Discussion by tnxz** [cirruslabs/tart#616] [Q&A] [answered]
any idea on how to enable shared directories with nixos
```
  fileSystems."/mnt" =
    { device = "share";
      fsType = "com.apple.virtio-fs.automount";
    };
```
tried this snippet with no success

**Comment by fkorotkov** [comment]
Hey @tnxz, could you please share a bit more details about your setup? We are not too familiar with running NixOS.

If you check `tart run --help` you can see that when you pass a shared directory via `--dir` flag to Linux VMs you need to manually mount it inside the VM by running something like `mount -t virtiofs com.apple.virtio-fs.automount /mnt/share`. Hope it helps and you can share either more information or if it will work for you than a working config so others can learn.

**Reply by tnxz** [reply]
I am just looking for a way to mount shared directories on boot in nixos with the above snippet

In nixos the following snippet in `configuration.nix`
```
  fileSystems."/" =
    { device = "/dev/disk/by-uuid/xxxxxx";
      fsType = "ext4";
    };
```
changes to this in `/etc/fstab`
```
/dev/disk/by-uuid/xxxxxx / ext4 ...
```

so what would `fsType` or `device` for the shared directories be

**Reply by edigaryev** [reply]
Try using `virtiofs` as an `fsType`, and `com.apple.virtio-fs.automount` as a `device`.

**Reply by tnxz** [reply]
that seems to work thanks

**Comment by tnxz** [comment, marked as answer]
```
  fileSystems."/mnt" =
    { device = "com.apple.virtio-fs.automount";
      fsType = "virtiofs";
    };
```
